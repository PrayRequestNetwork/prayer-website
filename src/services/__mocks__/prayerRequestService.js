export const mocks = {
  postPrayerRequest: jest.fn(),
};

export default class PrayerRequestServiceMock {
  postPrayerRequest = mocks.postPrayerRequest;
}