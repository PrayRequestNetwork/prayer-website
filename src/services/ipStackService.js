import superagent from 'superagent';

const ipStackBaseURI = 'http://api.ipstack.com/';

export default class IPStackService {

  fetchSingleAddress(address){
    
  }

  static request() {
    
  }
}