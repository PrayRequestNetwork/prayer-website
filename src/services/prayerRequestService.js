import superagent from 'superagent';

export default class PrayerRequestService {

  getAccountRoles = async () => {
    return await this.request({
      method: 'PUT',
      url: '/api/v1/user',
    });
  }

  getPrayers = async () => {
    return await this.request({
      method: 'GET',
      url: '/api/v1/prayer',
    });
  }

  delay(t, v) {
    return new Promise(function (resolve) {
      setTimeout(resolve.bind(null, v), t);
    });
  }
  
  postPrayerRequest = async payload => {
    return await this.request({
      method: 'POST',
      url: '/api/v1/prayer',
      payload,
    });
  }

  request = async ({method, url, payload}) => {
    url = `${process.env.REACT_APP_DEV_API_URL}${url}`;
    let result;
    if (!payload) {
      result = await superagent[method.toLowerCase()](url);
    } else {
      result = await superagent[method.toLowerCase()](url)
        .send(payload); 
    }
    return result.body;
  }
}