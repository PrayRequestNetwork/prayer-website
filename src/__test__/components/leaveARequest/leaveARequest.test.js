import React from 'react';
import { shallow } from 'enzyme';

import { mocks as PrayerRequestServiceMocks } from '../../../services/prayerRequestService';

import LeaveARequest from '../../../components/leaveARequest/leaveARequest';
import Form from '../../../components/forms/homepageForm/HomepageForm';
import Message from '../../../components/message/message';

jest.mock('../../../services/prayerRequestService');

describe('LeaveARequest', () => {
  it('Should Render LeaveARequest Component', () => {
    const wrapper = shallow(<LeaveARequest />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('Should Render Form Component', () => {
    const wrapper = shallow(<LeaveARequest />);
    const form = wrapper.find(Form);
    expect(form).toBeDefined();
  });

  describe('Handle Submit', () => {
    describe('On Success', () => {

      beforeEach(() => {
        PrayerRequestServiceMocks.postPrayerRequest.mockReturnValueOnce(
          new Promise(resolve => resolve('success'))
        );
      });

      it('Should toggle postingSuccess and displayMessage if PrayerRequestService returns success', async () => {
        const wrapper = shallow(<LeaveARequest />);
        const form = wrapper.find(Form);
        await form.props().handleOnSubmit({ prayerRequestMessage: 'TEST' });
        expect(wrapper.state().postingSuccess).toBe(true);
        expect(wrapper.state().displayMessage).toBe(true);
      });
      
      it('Should render message content with success text', async () => {
        const wrapper = shallow(<LeaveARequest />);
        const form = wrapper.find(Form);
        await form.props().handleOnSubmit({ prayerRequestMessage: 'TEST' });
        const message = wrapper.find(Message);
        const h4 = message.find('h4');
        const p = message.find('p');
        expect(message.debug()).toMatchSnapshot();
        expect(h4.hasClass('success-title')).toEqual(true);
        expect(p.hasClass('success-body')).toEqual(true);
      });
    });
    
    describe('On Failure', () => {

      beforeEach(() => {
        PrayerRequestServiceMocks.postPrayerRequest.mockReturnValueOnce(
          new Promise(resolve => resolve('error'))
        );
      });
      
      it('Should toggle postingFailure and displayMessage if PrayerRequestService returns error', async () => {
        const wrapper = shallow(<LeaveARequest />);
        const form = wrapper.find(Form);
        await form.props().handleOnSubmit({ prayerRequestMessage: 'TEST' });
        expect(wrapper.state().postingFailure).toBe(true);
        expect(wrapper.state().displayMessage).toBe(true);
      });
        
      it('Should render message content with failure text', async () => {
        const wrapper = shallow(<LeaveARequest />);
        const form = wrapper.find(Form);
        await form.props().handleOnSubmit({ prayerRequestMessage: 'TEST' });
        const message = wrapper.find(Message);
        const h4 = message.find('h4');
        const p = message.find('p');
        expect(message.debug()).toMatchSnapshot();
        expect(h4.hasClass('failure-title')).toEqual(true);
        expect(p.hasClass('failure-body')).toEqual(true);
      });
    });
  });
});