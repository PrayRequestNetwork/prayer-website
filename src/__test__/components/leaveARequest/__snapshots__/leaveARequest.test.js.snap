// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaveARequest Handle Submit On Failure Should render message content with failure text 1`] = `
"<Message showHandler={[Function]}>
  <div>
    <h4 className=\\"failure-title\\">
      We did not recieve your request.
    </h4>
    <p className=\\"failure-body\\">
      Something went wrong. Please try to submit again. If the problem continues, try refreshing the page and try again.
    </p>
  </div>
</Message>"
`;

exports[`LeaveARequest Handle Submit On Success Should render message content with success text 1`] = `
"<Message showHandler={[Function]}>
  <div>
    <h4 className=\\"success-title\\">
      We have successfully received your prayer request.
    </h4>
    <p className=\\"success-body\\">
      Our prayer team has been emailed and we will be praying for you. Although we do not know you, we know that our God knows you and each one of your concerns. We hope you will be blessed knowing that, along with Christ, there are others interceding on your behalf.
    </p>
  </div>
</Message>"
`;

exports[`LeaveARequest Should Render LeaveARequest Component 1`] = `
"<div id=\\"leaveARequest\\" className=\\"leaveARequest\\">
  <Form handleOnSubmit={[Function]} submitDisabled={false} />
</div>"
`;
